src/cache.rs

```diff
@@ -305,7 +305,7 @@ impl Cache {
         let mut pages = WalkDir::new(platforms_dir)
             .min_depth(1) // Skip root directory
             .into_iter()
-            .filter_entry(|e| should_walk(e)) // Filter out pages for other architectures
+            .filter_entry(|e| self.platform.is_all() || should_walk(e)) // Filter out pages for other architectures
```

Comment:
This check / the closure _could_ move into a potential `PlatformStrategy` :D

Reply:
I don't think the platform strategy should deal with file and directory names.

Reply:
Fair